ext/standard/basic_functions.stub.php

```diff
@@ -1098,8 +1054,7 @@ function intdiv(int $dividend, int $divisor): int {}
 
 function is_infinite(float $number): bool {}
 
-/** @return mixed */
-function pow($base, $exp) {}
+function pow($base, $exp): int|float {}
```

Comment:
I believe this is not strictly true ... do we have any tests that call pow() on GMP objects?

Reply:
> do we have any tests that call pow() on GMP objects?

No, we don't currently have any such test. Therefore, I've just added one.

> I believe this is not strictly true

Uh, I see now how this is possible (GMP defines its own handler).

Would `int|float|object` suffice then? Or should I remove the return type altogether since other extensions (not just built-in ones) can also define their own custom handler?

Reply:
We type this as `MAY_BE_NULL | MAY_BE_LONG | MAY_BE_DOUBLE | MAY_BE_OBJECT` in opcache, so I guess that would be fine? (The null cases is probably outdated).

Reply:
That makes sense! However, `gmp_pow()` doesn't support negative exponents, in which case it returns `false` + warning. So I'll promote this to an exception.